src/pet.js

```diff
@@ -0,0 +1,239 @@
+const {
+  speciesList,
+  specialAbilities,
+  lowHealthMessage,
+  tooFat,
+  needsFood,
+  checkUpMessage,
+  speciesData,
+  babySayings,
+} = require("./speciesList.js");
+
+const { MIN_XP_BY_LEVEL, MAX_XP_BY_LEVEL, MAX_LEVEL } = require("./levels.js");
+
+// Min and Max Stats //
+const MAX_FITNESS = 10;
+const MIN_FITNESS = 0;
+
+const MAX_HUNGER = 20;
+const MIN_HUNGER = 0;
+
+function Alien(name) {
+  this.name = name;
+  this.species = speciesList[Math.floor(Math.random() * speciesList.length)];
+  this.level = 20;
+  this.health = speciesData[this.species].health;
+  this.XP = 0;
+  this.hunger = 0;
+  this.fitness = speciesData[this.species].fitness;
+  this.specialAbilities = [speciesData[this.species].ability];
+  this.children = [];
+}
+
+Alien.prototype.isAlive = function () {
+  return this.level < MAX_LEVEL || this.health === 0;
+};
+
+Alien.prototype.levelUp = function () {
+  // Check level against the xp //
+  let minXP = MIN_XP_BY_LEVEL[this.level + 1];
+  let maxXP = MAX_XP_BY_LEVEL[this.level];
+  // Check if the player has enough XP to reach the next level
+  if (!this.XP >= minXP || !this.XP < maxXP) {
+    return console.log("Not enough XP to level up");
+  }
+
+  // Increment the level property by 1
+  this.level++;
+  // Decrement the fitness property by 3
+  this.fitness -= 3;
+
+  // If the level is divisible by 5, assign a new ability to the alien
+  if (this.level % 5 === 0) {
+    // Generate a random number between 0 and 19
+    let randomNumber = Math.floor(Math.random() * 20);
+
+    // Use the random number to select a random ability from the specialAbilities array
+    let ability = specialAbilities[randomNumber];
+
+    // Assign the ability to the alien
+    this.specialAbilities.push(ability);
+  }
+
+  // Generate a random number between 1 and 20
+  let health_increase = Math.floor(Math.random() * 20) + 1;
+
+  // Increase the alien's health by the health increase
+  this.health += health_increase;
+
+  // If the fitness level is below or equal to 0, set it to 0
+  if (this.fitness <= MIN_FITNESS) {
+    this.fitness = MIN_FITNESS;
+  }
+
+  console.log(this.name + " has leveled up!");
+
+  // Print the stats of the current instance of the Alien class
+  console.log("Here are " + this.name + "'s stats:");
+  console.log("Level: " + this.level);
+  console.log("HP: " + this.health);
+  console.log("XP: " + this.XP);
+  console.log("Hunger: " + this.hunger);
+  console.log("Fitness: " + this.fitness);
+
+  // Increment the hunger property by 5, but not if it is already at its maximum value of 20
+  if (this.hunger < MAX_HUNGER) {
+    this.hunger += 5;
+  }
```

Comment:
This comment is misleading, what if the hunger was set at 17? Current logic would update it to to 22. A cleaner way of doing this is:
```
  this.hunger = Math.min(this.hunger + 5, MAX_HUNGER);
}
```